kivy/uix/textinput.py

```diff
@@ -659,8 +659,22 @@ def insert_text(self, substring, from_undo=False):
                 int_pat = self._insert_int_patu
 
             if mode == 'int':
+                try:
+                    if self.text[0] == '-':
+                        substring = re.sub('-', '', substring)
+                    else:
+                        substring = re.sub('-', '', substring)
+                except:
+                    pass
```

Comment:
Why do you need the try/except? Also, both of the if/else cases have identical code...